subscription/internal/database/subscription_dao.go

```diff
@@ -0,0 +1,74 @@
+package database
+
+import (
+	"context"
+	"gorm.io/gorm"
+	"subscription-service/internal/app_errors"
+	"subscription-service/internal/models"
+	"time"
+)
+
+type SubscriptionRepository struct {
+	DB *gorm.DB
+}
+
+func NewSubscriptionRepository(db *gorm.DB) *SubscriptionRepository {
+	return &SubscriptionRepository{
+		DB: db,
+	}
+}
+
+func (d *SubscriptionRepository) Find(email string) (*models.Email, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*2)
+	defer cancel()
+
+	var subscription models.Email
+	result := d.DB.WithContext(ctx).Where("email = ?", email).Find(&subscription)
```

Comment:
As you fairly mentioned, we need do add some limits for pagination here

Reply:
and also, lets not use old branches for PRs, it is hard to focus and review such amount of changes. you can just not merge the PR in the master and keep it as a separate branch